sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2224,6 +2316,16 @@ object EliminateSubqueryAliases extends Rule[LogicalPlan] {
 }
 
 /**
+ * Removes [[View]] operators from the plan. The operator is respected till the end of analysis
+ * stage because we want to see which part of a analyzed logical plan is generated from a view.
+ */
+object EliminateView extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
+    case View(_, output, child) if child.isDefined => Project(output, child.get)
```

Comment:
This line can be simplied to `case View(_, output, Some(child)) => Project(output, child)`